tracex_project/trace_comparator/views.py

```diff
@@ -83,86 +112,154 @@ def get(self, request, *args, **kwargs):
         return super().get(request, *args, **kwargs)
 
     def post(self, request):
-        """Comparing a generated trace of a patient journey against the ground truth."""
-        patient_journey_name = self.request.session.get("patient_journey_name")
-        query_last_trace = Q(
-            id=Trace.manager.filter(patient_journey__name=patient_journey_name)
-            .latest("last_modified")
-            .id
-        )
-        pipeline_df = dfu.get_events_df(query_last_trace)
-        query_first_trace = Q(
-            id=Trace.manager.filter(patient_journey__name=patient_journey_name)
-            .earliest("last_modified")
-            .id
+        """Compare the newest trace of a patient journey against the ground truth and update session with results."""
+        patient_journey_name: str = self.request.session.get("patient_journey_name")
+        ground_truth_df, pipeline_df = self.get_first_and_last_trace(
+            patient_journey_name
         )
-        ground_truth_df = dfu.get_events_df(query_first_trace)
 
         try:
-            comparison_result_dict = compare_traces(self, pipeline_df, ground_truth_df)
+            comparison_result_dict: dict = compare_traces(
+                self, pipeline_df, ground_truth_df
+            )
         except Exception:  # pylint: disable=broad-except
             self.request.session.flush()
 
-            return render(self.request, "error_page.html", {"error_traceback": traceback.format_exc()})
+            return render(
+                self.request,
+                "error_page.html",
+                {"error_traceback": traceback.format_exc()},
+            )
 
         request.session["comparison_result"] = comparison_result_dict
 
         return redirect("testing_result")
 
 
-class TraceTestingResultView(TemplateView):
+class TraceTestingResultView(TemplateView, TraceComparisonMixin):
```

Comment:
what does this do? are why are we doing it?

Reply:
There is some logic that is used my multiple views. This Mixin allows both views to use this functionality without code duplication.